src/complex.jl

```diff
@@ -0,0 +1,355 @@
+export iscomplex,
+    isrealpart,
+    isimagpart,
+    isconj,
+    ordinary_variable,
+    degree_complex,
+    halfdegree,
+    mindegree_complex,
+    minhalfdegree,
+    maxdegree_complex,
+    maxhalfdegree,
+    extdegree_complex,
+    exthalfdegree
+
+"""
+    iscomplex(x::AbstractVariable)
+
+Return whether a given variable was declared as a complex-valued variable (also their conjugates are complex, but their real
+and imaginary parts are not).
+By default, all variables are real-valued.
+"""
+iscomplex(::AbstractVariable) = false
```

Comment:
Would it be possible to remove this function and use `!isreal` instead ? `iscomplex` is a bit ambiguous since the set of real numbers is a subset of the set of complex numbers. So in a sense, `iscomplex` should be true for real ones as well.

Reply:
While I fully understand your argument, isn't the issue the same with `isreal`? One could also assume that `isreal(x) = iszero(imag(x))`, but this is not the way it is implemented. Maybe this is even an argument for renaming (and therefore not using the Base function) them into `is_declared_real` and/or `is_declared_complex` or something similar?

Reply:
I thought some more about your comment. My previous response was more focused on pointing out the "for all values" character of these methods even more, but when applied to a raw variable, it should be quite clear that the behavior is indeed "for all values" - so adding unnecessary "declared" function name complexity does not make much sense.

My choice to have a function `iscomplex` that enables the complex-valued behavior came from two ideas:
- The default behavior is reals-only. So an additional function has to be provided that "enables" the behavior. This is different in mindset (but nothing else) to having a function that provides more of a legacy not-supported switch.
- I needed to do some special handling for the complex case. So actually, I rarely (actually never) use `isreal`, but `iscomplex` arises much more often. This is more of a convenience argument - but strictly speaking, `!isreal` is still shorter than `iscomplex`.

So if you wish, I'll remove `iscomplex` completely and make everything default to `!isreal`.